log.h

```diff
@@ -30,6 +30,11 @@
 #include <string>
 #include <vector>
 #include <mutex>
+#ifdef _WIN32
+// Include mingw-std-threads extra header
+#include <mingw.mutex.h>
```

Comment:
So mingw has two different threading "ABIs" for Windows: posix and native. The "posix" mode works by having a library that emulates the pthreads API using the Windows native threading capability whereas the "native" mode just doesn't have that. However, libstdc++ implements the C++11 threading interfaces using the pthreads API, so the "native" threading mode version of MinGW just doesn't have the C++ threads API available at all. So you now have two choices: a) use the "posix" version of the mingw compiler (and become dependent on the pthreads-win32 code) or b) use the "native" version of the mingw compiler (and become dependent on some other implementation of C++11 threads (which is mingw-std-threads in this case)). I somewhat-arbitrarily decided I liked option (b) better (despite the fact that yosys-abc still ends up requiring pthreads-win32 anyways).